packages/create-hydrogen/src/prompts/init.ts

```diff
@@ -1,40 +1,34 @@
-import {ui, github, string} from '@shopify/cli-kit'
+import {ui, github} from '@shopify/cli-kit'
 
 const TEMPLATE_BASE = 'https://github.com/Shopify/hydrogen/templates/'
 const BRANCH = `dist`
-const TEMPLATE_DATA = {
+const TEMPLATE_NAME_DATA = {
   /* eslint-disable @typescript-eslint/naming-convention */
-  'demo-store-js': {
+  'demo-store': {
     description: 'Demo Store',
   },
-  'demo-store-ts': {
-    description: 'Demo Store (TypeScript)',
-  },
-  'hello-world-js': {
+  'hello-world': {
     description: 'Hello World',
   },
-  'hello-world-ts': {
-    description: 'Hello World (TypeScript)',
-  },
-  /* eslint-enable @typescript-eslint/naming-convention */
 }
 
-const TEMPLATE_NAMES = Object.keys(TEMPLATE_DATA)
-
-function toHydrogenTemplateName(key: string) {
-  const normalized = string.hyphenize(key).toLocaleLowerCase()
-  const withExtension = normalized.endsWith('-ts') || normalized.endsWith('-js') ? normalized : `${normalized}-js`
-
-  return TEMPLATE_NAMES.includes(withExtension) ? withExtension : false
+const LANGUAGE_NAME_DATA = {
+  js: {
+    description: 'JavaScript',
```

Comment:
This sounds good - since I'll be rewriting some stuff to fix passing explicit template's through the CLI, might also look into throwing this in there too :)